### PR TITLE
Implement discussed API

### PR DESCRIPTION
**Pull request by burke**
This implements the API we just walked through.

One problem I discovered is that, now that the registry lives outside of the `Command` namespace, nothing forces registration of those commands anymore.

I'll have to find some way to cause CommandRegistry loading to load the Commands module.

### CODE REVIEW

**Comment by burke** [comment]
Second commit moves `Todo::CommandRegistry` to `Todo::Commands::Registry` to force commands to load when autoloading registry.

File: lib/todo.rb [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+  end
+
+  autocall(:ErrorHandler) do
+    CLI::Kit::ErrorHandler.new(
+      log_file: LOG_FILE,
+      exception_reporter: nil
+    )
+  end
+
+  Abort        = CLI::Kit::Abort
```
Comment (jules2689) [inline lib/todo.rb]: This is a bit :/ But also not... /shrug

File: lib/todo/commands.rb [hunk 2]
```diff
@@ -2,11 +2,16 @@
 
 module Todo
   module Commands
-    REGISTRY = CLI::Kit::CommandRegistry.new(default: 'help')
+    # No point in using autoload/autocall here; it's loaded immediately by
+    # `register` calls.
+    CLI::Kit::CommandRegistry.new(
```
Comment (jules2689) [inline lib/todo/commands.rb]: What is this assigned to? Confused what this actually accomplishes

File: lib/todo.rb (hunk 1 above)
Comment (burke) [reply lib/todo.rb]: Yeah, unsure about this one. We use this extensively in `dev` but ... I think it's a useful pattern?

File: lib/todo/commands.rb (hunk 2 above)
Comment (burke) [reply lib/todo/commands.rb]: refactor fail, fixed

File: lib/todo.rb (hunk 1 above)
Comment (jules2689) [reply lib/todo.rb]: Maybe we should just use `CLI::Kit::Abort` directly and not pretend like it's different?

File: lib/todo.rb [hunk 3]
```diff
@@ -1,18 +1,33 @@
...
+CLI::UI::StdoutRouter.enable
 
 module Todo
-  autoload :Config,       'todo/config'
-  autoload :Command,      'todo/command'
-  autoload :EntryPoint,   'todo/entry_point'
+  extend CLI::Kit::Autocall
+
+  TOOL_NAME = 'todo'
+  ROOT      = File.expand_path('../..', __FILE__)
```
Comment (lugray) [inline lib/todo.rb:10]: I don't think this is used?

Comment (burke) [reply lib/todo.rb:10]: No, but I anticipated putting this in the template because it's frequently useful.

**Pull request merged**